### PR TITLE
Added Golang

### PR DESCRIPTION
**Pull request by aminasadiam**

### CODE REVIEW

**Comment by BaseMax** [comment]
Hi Amin,
Welcome, honestly we reserved Go for Parham, but I aware he is a bit busy. So we can check your PR. Thanks.

---------

Hi Parham, Can you take a look and review it? @1995parham

File: GregorianToJalali.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package main
+
+import "fmt"
+
+type DateStruct struct {
```
Comment (1995parham) [inline GregorianToJalali.go:5]: I think here using `Date` is better than `DateStruct`.

File: GregorianToJalali.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	} else {
+		result.Day = 1 + (days-186)%30
+	}
+
+	return result
+}
+
+func GregorianToJalaliStr(year int, month int, day int) string {
+	result := GregorianToJalali(year, month, day)
+	return fmt.Sprintf("%d/%d/%d", result.Year, result.Month, result.Day)
```
Comment (1995parham) [inline GregorianToJalali.go:60]: we can have it as `String` method on `Date` struct.

File: GregorianToJalali.go (hunk 1 above)
Comment (aminasadiam) [reply GregorianToJalali.go:5]: hi Mr.Parham, I made this change.

File: GregorianToJalali.go (hunk 2 above)
Comment (aminasadiam) [reply GregorianToJalali.go:60]: sorry, i didn't get you🤔

Comment (aminasadiam) [reply GregorianToJalali.go]: I made this change to.

**Comment by BaseMax** [comment]
Please add more unit tests if possible. (`assert.Equal`)

**Comment by aminasadiam** [comment]
Done. I hope it's ok. @BaseMax

**Comment by BaseMax** [comment]
> Done. I hope it's ok. @BaseMax 

Hi Amin, thank you.
It's okay in my side. Pargam will comment and merge it.

Let me know if you want to add other languages too

Best
M.

**Comment by aminasadiam** [comment]
Glad to participate. I can also add typescript. But I think Mr. Erfan will do that.

**Review by BaseMax — CHANGES_REQUESTED** [review]
We convert the day and month digit to two number not one.
For example: "1" => "01"

But your commit:
```go
func (d Date) String() string {
	return fmt.Sprintf("%d/%d/%d", d.Year, d.Month, d.Day)
}
```

You can see this in other languages too:
- https://github.com/BaseMax/gregorian_to_jalali/blob/main/gregorian_to_jalali.c#L58
- https://github.com/BaseMax/gregorian_to_jalali/blob/main/gregorian_to_jalali.js#L50

**Comment by aminasadiam** [comment]
hi @BaseMax , I made this change.

**Comment by BaseMax** [comment]
Hi, I going merge it, but this has some problem which need to fix.

**Comment by BaseMax** [comment]
Check the commit and my change:
https://github.com/BaseMax/gregorian_to_jalali/commit/aabb5b401b100adeaf4a6e3efa6efcdc64b25a64

You will see `String` is how much easy and I removed all of your ifs.

**Comment by BaseMax** [comment]
By the way, the **test file** is not ready yet and need contribute.
if want to work on, its: https://github.com/BaseMax/gregorian_to_jalali/blob/main/GregorianToJalali.test.go#L9

**Pull request merged**